### DISCUSSION TITLE
High CPU usage on services.exe

### DISCUSSION THREAD

**Discussion by recepaslan** [quartznet/quartznet#1144] [Q&A] [answered]
Hi, 

Our background application which is built using Quartz causes `Services and Controller App (services.exe)` to use a high CPU. The application itself uses an acceptable range of CPU (0-1% when idle, 3-4% when jobs triggered) and memory (max 200MB). But whenever jobs are triggered, `services.exe` CPU usage gets increased and causes problems on the server such prevents other services to restart.

Some insights from the application:
* Around 75 jobs, 5-10 of them are 2 mins interval, the rest to run at midnight. (Issue happens all the time)
* `quartz.threadPool.threadCount = 10`
* `quartz.scheduler.batchTriggerAcquisitionMaxCount = 100`
* Uses RAMJobStore
* Used singleton pattern with IoC Container.

Any help appreciated.

**Comment by lahma** [comment, marked as answer]
I think you need to attach a profiler and see what is eating all the CPU, my guess is that your jobs are CPU heavy? Also might help enabling logging if there's some sort of retry happening or failures.

**Reply by recepaslan** [reply]
I will followup with retry or failures. 
Jobs are almost empty at this point, they are just doing SQL queries and exit.

**Reply by recepaslan** [reply]
Find the issue with `dotTrace`, and gladly I can say it is not related to Quartz :). Thanks for the quick response. 

I have asked the same question on StackOverflow and responded to my findings there. I know the title is wrong now, but it was just a starting point for me.

https://stackoverflow.com/questions/66817698/quartz-net-causes-services-exe-to-cause-cpu-spike-on-services-exe/66819190#66819190
